lib/roo_on_rails/logger.rb

```diff
@@ -0,0 +1,92 @@
+require 'logger'
+require 'delegate'
+require 'roo_on_rails/logfmt'
+
+module RooOnRails
+  class Logger < SimpleDelegator
+    def initialize(io = STDOUT)
+      @show_timestamp = io.tty?
+      logger = ::Logger.new(io).tap do |l|
+        l.formatter = method(:_formatter)
+      end
+      super(logger)
+      set_log_level
+    end
+
+    def with(context = {})
+      unless block_given?
+        return Proxy.new(self, context)
+      end
+
+      new_context = (_context_stack.last || {}).merge(context)
+      Thread.handle_interrupt(Exception => :never) do
+        begin
+          _context_stack.push(new_context)
+          Thread.handle_interrupt(Exception => :immediate) do
+            yield self
+          end
+        ensure
+          _context_stack.pop
+        end
+      end
+      nil
+    end
+
+    def clear_context!
+      _context_stack.replace([{}])
+    end
+
+    def set_log_level
+      self.level = ::Logger::Severity.const_get(ENV.fetch('LOG_LEVEL', 'DEBUG'))
+    end
+
+    private
+
+    class Proxy < SimpleDelegator
```

Comment:
Nit: maybe move the inner class before the `private` block, since it isn't really private? Or maybe you want make it private. Either way, the intention here is a bit unclear.

Reply:
It's intended to be private, hence placing it here — event though as you point out Ruby doesn't have private subclasses.

Reply:
It kind of does - you can use `private_constant`.